lib/web/apiserver.go

```diff
@@ -1772,6 +1777,46 @@ func (h *Handler) changeUserAuthentication(w http.ResponseWriter, r *http.Reques
 	}, nil
 }
 
+func (h *Handler) trySettingConnectorNameToPasswordless(ctx *SessionContext, r *http.Request, req changeUserAuthenticationRequest) error {
+	if req.WebauthnCreationResponse == nil {
```

Comment:
can we tell if they picked passwordless? What happens if they register a webauthn device as a second factor? What happens if they initially sign in with TOTP and then add a passwordless device later?

Reply:
> can we tell if they picked passwordless?
Yeah, If a user submits a passwordless form the `req.WebauthnCreationResponse` field will contain data. In case the user picks password + TOTP the `req.SecondFactorToken` field will contain token.
> What happens if they register a webauthn device as a second factor?
The auth preference will remain as is. In case user adds second factor the other endpoint is called: `POST /webapi/mfa/devices`

Reply:
>> can we tell if they picked passwordless? What happens if they register a webauthn device as a second factor?

> Yeah, If a user submits a passwordless form the req.WebauthnCreationResponse field will contain data

We are baking in an UI assumption here, we **cannot** tell the difference between "regular" WebAuthn and Passwordless just by looking at the CredentialCreationResponse. It would be better to get a clear signal, from the UI, that the user indeed went through the passwordless flow.